tz/hash.go

```diff
@@ -67,9 +60,9 @@ func NewWith(impl Implementation) hash.Hash {
 
 // New returns a new hash.Hash computing the Tillich-Zémor checksum.
 func New() hash.Hash {
-	if hasAVX2 {
+	if cpu.X86.HasAVX2 {
```

Comment:
cpu.X86.HasAVX2 && cpu.X86.HasAVX

Reply:
see [`x/sys` package](https://github.com/golang/sys/blob/86b910548bc16777f40503131aa424ae0a092199/cpu/cpu_x86.go#L50)

Reply:
I don't think that's enough, because our AVX2 implementation uses AVX instructions, not only X*/Y* registers.